### DISCUSSION TITLE
How to select authentication provider by url (query string)

### DISCUSSION THREAD

**Discussion by jonaskello** [requarks/wiki#6089] [Help / Questions] [answered]
We have a product documented in wikijs. This product is used by several customers. Each customer has their own identity provider (login server). So in wikijs we can register separate authentication strategies for each customer (separate login servers). However now the wikijs login page shows a long list of different login servers and we don't want customers to see login servers of other customers.

Each customer is running a separate instance of the product and we link to the wiki from within the product. So we could have a custom entry url for each customer into the wiki, something like:

`https://wiki.product.com?idp=customer1`
`https://wiki.product.com?idp=customer2`
`https://wiki.product.com?idp=customer3`

Then wikijs could directly pick the identity provider (authentication strategy) based on the `idp` query string paramter. This idea comes from keycloak which has a query string parameter called [`kc_idp_hint`](https://www.keycloak.org/docs/latest/server_admin/#default_identity_provider) which lets you specifiy which identity provider keycloak should automatically use.

Is there any way to have wikijs do something like this? Or some other way to solve the original problem?

**Comment by NGPixel** [comment, marked as answer]
You could link to the auth strategy login route directly? e.g.:

```
https://wiki.product.com/login/d44d5a51-44ad-484c-acd3-8d1a98f14d77
```

where `d44d5a51-44ad-484c-acd3-8d1a98f14d77` is the UUID of the auth strategy (look for the ID in the admin).

**Reply by jonaskello** [reply]
Thank you, that works great! :-)
